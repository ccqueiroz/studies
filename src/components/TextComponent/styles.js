import styled, { css } from 'styled-components';

const transformTextEllipsis = (textEllipsis) => {
  return textEllipsis
    ? css`
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    `
    : css`
      text-overflow: unset;
      white-space: unset;
      overflow: unset;
    `;
};

export const Container = styled.p`
  ${({
    theme,
    lineHeight,
    letterSpacing,
    textIndent,
    textAlign,
    wordBreak,
    textEllipsis,
  }) => css`
    font-size: ${theme.font.sizes.medium};
    line-height: ${lineHeight};
    letter-spacing: ${letterSpacing};
    text-indent: ${textIndent};
    text-align: ${textAlign};
    word-break: ${wordBreak ? 'break-word' : 'unset'};
    ${transformTextEllipsis(textEllipsis)};
  `}
`;
