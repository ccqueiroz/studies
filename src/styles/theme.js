export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
  },
  font: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      xsmall: '0.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
    letterSpacing: {
      none: 'unset',
      xsmall: '0.1rem',
      small: '0.2rem',
      medium: '0.3rem',
      large: '0.5rem',
      xlarge: '0.75rem',
      huge: '0.9rem',
      xhuge: '1rem',
    },
  },
  media: {
    ltMedium: '(max-width: 768px)',
  },
};
