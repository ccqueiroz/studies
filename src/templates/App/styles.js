import style, { css } from 'styled-components';

export const Wrapper = style.div`
  ${(props) => css`
    background: ${props.background};
  `}
`;
