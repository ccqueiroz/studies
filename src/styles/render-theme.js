import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderThemeProvider = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};
