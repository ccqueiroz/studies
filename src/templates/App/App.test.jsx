import { screen } from '@testing-library/react';
import { renderThemeProvider } from '../../styles/render-theme';
import Home from '.';

describe('[Component]: <Home />', () => {
  it('should be render Home correctly', () => {
    renderThemeProvider(<Home />);
    const headingContainer = screen.getByRole('heading', {
      name: /Hello World!/i,
    }).parentElement; //parentElement busca o pai do container
    expect(headingContainer).toHaveStyle({
      background: 'red',
    });
  });
  it('should be render snapshot Home', () => {
    renderThemeProvider(<Home />);
    const headingContainer = screen.getByRole('heading', {
      name: /Hello World!/i,
    }).parentElement; //parentElement busca o pai do container
    expect(headingContainer).toMatchSnapshot();
  });
});
