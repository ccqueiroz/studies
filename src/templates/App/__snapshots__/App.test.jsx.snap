// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: <Home /> should be render snapshot Home: 
      .c0 {
        background: red;
      }

      <div
        class="c0"
      >
        <h1>
          Hello World!
        </h1>
      </div>
     1`] = `
.c0 {
  background: red;
}

<div
  class="c0"
>
  <h1>
    Hello World!
  </h1>
</div>
`;
